convert2rhel/actions/__init__.py

```diff
@@ -206,25 +211,122 @@ def run(self):
 
         self._has_run = True
 
-    def set_result(self, status=_NO_USER_VALUE, error_id=_NO_USER_VALUE, message=_NO_USER_VALUE):
+    @property
+    def result(self):
+        """
+        Helper method to save result variable in a different variable, so it does
+        not get overwritten by the __init__ method of the Action class
+        """
+        return self._result
+
+    @result.setter
+    def result(self, action_message):
+        """
+        Make sure the correct policy for result messages is used.
         """
-        Helper method that sets the resulting values for status, error_id and message.
+        if not isinstance(action_message, ActionResult):
+            raise Exception()
```

Comment:
This should raise a specific exception instead of a generic `Exception`.  `TypeError` is probably appropriate since we're expecting a specific type of parameter, an `ActionResult`.

Reply:
I will change this